src/main/java/com/google/cloud/tools/maven/configs/module/DefaultListConfigurationImpl.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.maven.configs.module;
+
+import com.google.cloud.tools.app.Option;
+import com.google.cloud.tools.app.config.module.ListConfiguration;
+import com.google.cloud.tools.maven.configs.GcpAppGoalConfiguration;
+import com.google.cloud.tools.maven.module.List;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Configuration for the {@link List} goal.
+ */
+public class DefaultListConfigurationImpl extends GcpAppGoalConfiguration
+    implements ListConfiguration {
+
+  private static String MODULES_KEY = "gcp.app.module.list.modules";
+  private static String SERVER_KEY = "gcp.app.module.list.server";
+
+  private Collection<String> modules = ImmutableList.of();
+  private String server;
+
+  public Collection<String> getModules() {
+    return modules;
+  }
+
+  public Map<Option, String> getOptionalParameters() {
```

Comment:
Sorry, I misunderstood what was going on here. I think creating the optional parameters map here bypasses the advantages of using the ListConfiguration interface model.

What do you think about putting this processing in ListAction? So that anyone making a ListConfiguration can passively reuse this?